File: barras.py
import pandas as pd
import matplotlib.pyplot as plt

# Substitua "caminho/para/seu/arquivo.csv" pelo caminho do arquivo CSV em sua máquina
# data = pd.read_csv("caminho/para/seu/arquivo.csv")

# A linha 5 substitui essa, pois aqui estou criando dados fictícios
# Define um dicionário com os dados das classes de veículo, pois não estou usando nenhum arquivo CSV
data = {
    'class': ['compact', 'midsize', 'suv', 'midsize', 'compact', 'suv', 'suv', 'compact'],
}

# Caso você já tenha a variável de frequência calculada pode pular para a linha 25
# Cria um DataFrame chamado 'mpg' com os dados fornecidos
mpg = pd.DataFrame(data)

# Conta o número de ocorrências de cada classe de veículo e armazena em 'class_counts'
class_counts = mpg['class'].value_counts().reset_index()

# Renomeia as colunas para 'class' e 'count'
class_counts.columns = ['class', 'count']

# Ordena as contagens pelo número de ocorrências
# Se a sua variável frequência já estiver calculada:
# data = data.sort_values(by='frequencia')
class_counts = class_counts.sort_values(by='count')

# Criar o gráfico de barras
plt.figure(figsize=(10, 6))
# class é a variável x e count é a variável y
plt.bar(class_counts['class'], class_counts['count'], color='#6E9AF8')
plt.title('Distribuição das classes de veículo')
plt.xlabel('Classe')
plt.ylabel('Frequência')
plt.xticks(rotation=45)
plt.show()

File: barras.R
library(dplyr)
library(ggplot2)
library(hrbrthemes)
library(readr)

# Substitua "caminho/para/seu/arquivo.csv" pelo caminho do arquivo CSV em sua máquina
# mpg <- read.csv("caminho/para/seu/arquivo.csv", header = TRUE)

# Criar o gráfico de barras
 mpg %>%
    count(class) %>% # Substitua class pela sua variável categórica
    # Caso exista, substitua n pela sua variável de frequência
    mutate(class = reorder(class, n)) %>% # Ordenando a variável pela frequência
    ggplot(aes(x = class, y = n)) + 
    geom_bar(fill = "#6E9AF8", stat = "identity") + 
    ggtitle("Distribuição das classes de veículo") +
    xlab("classe") +
    ylab("frequência") +
    theme_ipsum()
    theme(axis.text.x = element_text(size = 14),
          axis.text.y = element_text(size = 14),
          axis.title.x = element_text(size = 14),
          axis.title.y = element_text(size = 14))
